sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -764,3 +748,146 @@ def update_repository_properties(self, *args, **kwargs):
+    ) -> None:
+        """Upload a manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param manifest: The manifest to upload.
+        :type manifest: ~azure.containerregistry.models.OCIManifest or IO
+        :keyword tag: Tag of the manifest.
+        :paramtype tag: str or None
+        :returns: None
+        :rtype: None
+        :raises ValueError: If the parameter repository or manifest is None.
+        """
+        if (repository or manifest) is None:
+            raise ValueError("The parameter repository and manifest cannot be None.")
+
+        tag_or_digest = tag
+        if tag is None:
+            stream = manifest
+            if isinstance(manifest, OCIManifest):
+                stream = _serialize_manifest(manifest)
+            tag_or_digest = _compute_digest(stream)
+
+        self._client.container_registry.create_manifest(
+            name=repository, reference=tag_or_digest, payload=stream, content_type=OCI_MANIFEST_MEDIA_TYPE, **kwargs
+        )
+
+    @distributed_trace
+    def upload_blob(self, repository, stream, **kwargs):
+        # type: (str, IO, **Any) -> None
+        """Upload an artifact blob.
+
+        :param str repository: Name of the repository
+        :param stream: The manifest to upload.
+        :type stream: IO
+        :returns: None
+        :rtype: None
+        :raises ValueError: If the parameter repository or stream is None.
+        """
+        if (repository or stream) is None:
+            raise ValueError("The parameter repository and stream cannot be None.")
+
+        start_upload_respose = self._client.container_registry_blob.start_upload(
+            repository, cls=_return_response, **kwargs
+        )
+        upload_chunk_response = self._client.container_registry_blob.upload_chunk(
+            start_upload_respose['Location'], stream, cls=_return_response, **kwargs
+        )
+        digest = _compute_digest(stream)
+        self._client.container_registry_blob.complete_upload(
+            digest=digest, next_link=upload_chunk_response['Location'], value=stream, cls=_return_response, **kwargs
+        )
+
+    @distributed_trace
+    def download_manifest(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> ManifestWrapper
+        """Download the manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param str tag_or_digest: The manifest to upload.
+        :returns: ManifestWrapper
+        :rtype: ~container_registry.models.ManifestWrapper
+        :raises ValueError: If the parameter repository or tag_or_digest is None.
+        """
+        if repository or tag_or_digest is None:
+            raise ValueError("The parameter repository and tag_or_digest cannot be None.")
+
+        return self._client.container_registry.get_manifest(
+            name=repository, reference=tag_or_digest, accept=OCI_MANIFEST_MEDIA_TYPE, **kwargs
```

Comment:
How big will this be? i.e. will this method block while downloading a large payload?

Reply:
Followed other languages, not sure whether will block while the payload is large.